.github/workflows/publish.yml

```diff
@@ -11,34 +11,78 @@ on:
     paths:
       - "src/**"
       - ".github/workflows/publish.yml"
+  repository_dispatch:
+    types: [create-release]
+    client_payload:
+      release_tag:
+        required: true
 
 jobs:
+  create_release:
+    if: ${{ github.event_name == 'repository_dispatch' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: Brightspace/third-party-actions@actions/checkout
+      - name: Create release
+        env:
+          RELEASE_TAG: ${{ github.event.client_payload.release_tag }}
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          gh release create "$RELEASE_TAG" --draft --generate-notes --title "Release $RELEASE_TAG"
+
   build:
+    needs: create_release
+    if: ${{ !failure() }}
     strategy:
       matrix:
         include:
           - machine: windows-latest
             platform: win
+            file_name: bmx.exe
           - machine: macos-latest
             platform: osx
+            file_name: bmx
     runs-on: ${{ matrix.machine }}
     timeout-minutes: 10
+    env:
+      PLATFORM: ${{ matrix.platform }}
     steps:
       - name: checkout
         uses: Brightspace/third-party-actions@actions/checkout
+      - name: install 7zip
+        if: matrix.platform == 'win'
+        run: choco install 7zip -y
+        shell: bash
       - name: publish
         shell: pwsh
+        working-directory: src/D2L.Bmx
+        run: |
+          dotnet publish -r $env:PLATFORM-x64 -c Release -o build
+      - name: upload build
+        if: ${{ github.event_name == 'repository_dispatch' }}
+        shell: bash
         env:
+          RELEASE_TAG: ${{ github.event.client_payload.release_tag }}
           PLATFORM: ${{ matrix.platform }}
+          FILE_NAME: ${{ matrix.file_name }}
+          GH_TOKEN: ${{ github.token }}
         working-directory: src/D2L.Bmx
-        run: dotnet publish -r $env:PLATFORM-x64 -c Release -o out
+        run: |
+          cd build
+          7z a "$PLATFORM-$FILE_NAME.zip" "$FILE_NAME"
+          gh release upload "$RELEASE_TAG" "$PLATFORM-$FILE_NAME.zip"
```

Comment:
nit: personally I'd prefer `bmx-{platform}.zip` rather than `{platform}-bmx.zip`.

When I download a tool onto my machine, the first thing I'm going to search for in my filesystem is the tool's name, e.g. "bmx". I won't be thinking about my "platform" - all my tools will be for my computer's platform. So putting the tool name "bmx" first would have more distinguishing power, e.g. good for auto complete on in the terminal.

Reply:
also, I don't think we should put ".exe" in the windows artifacts' name. we can just hardcode "bmx" here.